src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -2320,6 +2320,87 @@ .locals init (object[] V_0)
         verifier.VerifyIL("C.M", expectedIl);
     }
 
+    [Fact]
+    public void OverloadResolution_SpanVsIEnumerable()
+    {
+        var source = """
+            using System;
+            using System.Collections.Generic;
+
+            var a = new int[0];
+            C.M(a);
+
+            static class C
+            {
+                public static void M(Span<int> x) => Console.Write(1);
+                public static void M(IEnumerable<int> x) => Console.Write(2);
+            }
+            """;
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (5,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(Span<int>)' and 'C.M(IEnumerable<int>)'
+            // C.M(a);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(System.Span<int>)", "C.M(System.Collections.Generic.IEnumerable<int>)").WithLocation(5, 3));
```

Comment:
I'm not understanding why this is an error? `M(Span<int>)` should be an implicit span conversion, no?

Reply:
Yes, exactly, that's why this error appears only in C# 12 and not in later versions.

Reply:
Ah, right. Sorry, I read too fast.